Rename CODE_OWNERS -> Maintainers

AaronBallman opened the pull request.

AaronBallman committed:
Rename CODE_OWNERS -> Maintainers

AaronBallman commented:
I've kept everyone in the same position they were previously in, just reworded for the newer nomenclature.

That said, I'd like explicit confirmation from @r4nt @njames93 and @sam-mccall that they're still willing and able to volunteer under the new maintainers guidance (https://llvm.org/docs/DeveloperPolicy.html#maintainers) before landing these changes. If you're not able to do so, that's totally fine! I can move you to the inactive maintainers part of the list and we can try to find other volunteers to step in to the role.

Also, I cannot seem to find a GitHub handle for Julie Hockett and her last community interactions seem to be from 2021. If someone can help find the correct information and tag her on the review, that would be appreciated.

njames93 commented:
I just no longer have the time needed to hold maintainer status. I'll still contribute as and when I have the time.

AaronBallman commented:
> I just no longer have the time needed to hold maintainer status. I'll still contribute as and when I have the time.

Thank you for your help! I'll move you to the inactive maintainers list, and if you find you have more time in the future to step back into a maintenance role, we'll happily move you back to active status.

AaronBallman committed:
Move Nathan to the inactive maintainers section

Based on review feedback from Nathan -- thank you for all the help!

AaronBallman commented:
Ping @r4nt and @sam-mccall

AaronBallman committed:
Add Congcong Cai as a maintainer for clang-tidy

AaronBallman committed:
Add Julian Schmidt as a maintainer for clang-tidy

5chmidti reviewed (approved):
LGTM in regard to my details.

There are some emails that have their `@` escaped. Was that intended?

AaronBallman committed:
No need to escape the @

AaronBallman committed:
Add maintainers for clang-doc and clangd

ilovepi reviewed (approved):
Thanks! My details seem correct.

AaronBallman committed:
Add Manuel as an emeritus lead maintainer, add myself as the new lead

Manuel has been unresponsive on the PR and the post on Discourse, but
he had previously stepped down as a component maintainer in Clang due
to a lack of bandwidth, so I think it's safe to move him to the
inactive list.

If he wants to be lead again, I'll happily pass the duties back to him.

PeterChou1 commented on clang-tools-extra/Maintainers.txt:
```diff
@@ -0,0 +1,72 @@
+| paulkirth@google.com (email), ilovepi (GitHub), ilovepi (Discourse)
+
+| Peter Chou
+| peter.chou@mail.utoronto.ca (email), PeterChou1 (GitHub), PeterChou1 (Discourse), .peterchou (Discord)
```
my email changed since I graduated my new email is 
peterchou411@gmail.com

AaronBallman replied:
Thanks! I've corrected this.

AaronBallman committed:
Update email address for Peter Chou

5chmidti commented:
@PiotrZSL The ping was in case you want to be added as a maintainer to clang-tidy as well. I’m mentioning it just to make sure.

PiotrZSL commented:
> @PiotrZSL The ping was in case you want to be added as a maintainer to clang-tidy as well. I’m mentioning it just to make sure.

The designation of "maintainer" does not impact the established process. This role primarily introduces additional responsibilities without offering any demonstrable advantages. It is likely to become dead in the near future, mirroring the fate of the code owner role. At the end, being or not being a maintainer has no impact on me.

HerrCai0907 commented:
> At the end, being or not being a maintainer has no impact on me.
I think it will help the first time contributor in this sub project to find correct one who can  review their pr. But it is definitely not a  mandatory thing.

AaronBallman commented:
> > @PiotrZSL The ping was in case you want to be added as a maintainer to clang-tidy as well. I’m mentioning it just to make sure.
> 
> The designation of "maintainer" does not impact the established process.

It does -- it helps newcomers figure out who can help them with reviews, answering questions, etc.

> This role primarily introduces additional responsibilities without offering any demonstrable advantages.

The only advantage to the maintainer is to get published credit for their increased responsibilities in the community, which isn't an advantage for everyone. The primary advantage is for the community.

> It is likely to become dead in the near future, mirroring the fate of the code owner role. 

The project has had this same concept in some form or fashion for at least 15 years now, so I don't think this is an accurate view.

> At the end, being or not being a maintainer has no impact on me.

There's no pressure to sign up to be one, but having a dedicated group of listed, active maintainers makes it more likely the project will thrive. FWIW, projects without active maintainers are starting to be removed or subsumed and I expect we'll continue to do that into the future.

PiotrZSL commented on clang-tools-extra/Maintainers.txt:
```diff
@@ -0,0 +1,72 @@
+
+| Julian Schmidt
+| git.julian.schmidt@gmail.com (email), 5chmidti (GitHub), 5chmidti (Discourse), 5chmidti (Discord)
+
```
```suggestion

| Piotr Zegar
| me@piotrzegar.pl (email), PiotrZSL (GitHub), PiotrZSL (Discourse), PiotrZSL (Discord)

```

PiotrZSL reviewed (commented):
Hmm, Then you may consider adding me.

AaronBallman committed:
Update clang-tools-extra/Maintainers.txt

Co-authored-by: Piotr Zegar <me@piotrzegar.pl>
